scripts/camera/orbit-camera.mjs

```diff
@@ -0,0 +1,709 @@
+            );
+        }
+    }
+
+    get focusEntity() {
+        return this._focusEntity;
+    }
+
+    /**
+     * Frames the entity or scene at the start of the application."
+     *
+     * @attribute
+     * @title Frame on Start
+     * @type {boolean}
+     */
+    set frameOnStart(value) {
+        this._frameOnStart = value;
+        if (this._frameOnStart) {
+            this.focus(this.focusEntity || this.app.root);
+        }
+    }
+
+    get frameOnStart() {
+        return this._frameOnStart;
+    }
+
+    /**
+     * Property to get and set the distance between the pivot point and camera.
+     * Clamped between this.distanceMin and this.distanceMax
+     *
+     * @type {number}
+     */
+    set distance(value) {
+        this._targetDistance = this._clampDistance(value);
+    }
+
+    get distance() {
+        return this._targetDistance;
+    }
+
+    /**
+     * Property to get and set the camera orthoHeight
+     *
+     * @type {number}
+     */
+    set orthoHeight(value) {
+        this.entity.camera.orthoHeight = Math.max(0, value);
+    }
+
+    get orthoHeight() {
+        return this.entity.camera.orthoHeight;
+    }
+
+
+    /**
+     * Property to get and set the pitch of the camera around the pivot point (degrees).
+     * Clamped between this.pitchAngleMin and this.pitchAngleMax.
+     * When set at 0, the camera angle is flat, looking along the horizon
+     *
+     * @type {number}
+     */
+    set pitch(value) {
+        this._targetPitch = this._clampPitchAngle(value);
+    }
+
+    get pitch() {
+        return this._targetPitch;
+    }
+
+
+    /**
+     * Property to get and set the yaw of the camera around the pivot point (degrees)
+     *
+     * @type {number}
+     */
+    set yaw(value) {
+        this._targetYaw = value;
+
+        // Ensure that the yaw takes the shortest route by making sure that
+        // the difference between the targetYaw and the actual is 180 degrees
+        // in either direction
+        const diff = this._targetYaw - this._yaw;
+        const reminder = diff % 360;
+        if (reminder > 180) {
+            this._targetYaw = this._yaw - (360 - reminder);
+        } else if (reminder < -180) {
+            this._targetYaw = this._yaw + (360 + reminder);
+        } else {
+            this._targetYaw = this._yaw + reminder;
+        }
+    }
+
+    get yaw() {
+        return this._targetYaw;
+    }
+
+    /**
+     * Property to get and set the world position of the pivot point that the camera orbits around
+     *
+     * @type {number}
+     */
+    set pivotPoint(value) {
+        this._pivotPoint.copy(value);
+    }
+
+    get pivotPoint() {
+        return this._pivotPoint;
+    }
+
+
+    /** @private */
+    _distance = 0;
+
+    /** @private */
+    _distanceMin = 0;
+
+    /** @private */
+    _distanceMax = 0;
+
+    /** @private */
+    _frameOnStart = true;
+
+    /** @private */
+    _focusEntity;
+
+    static fromWorldPoint = new Vec3();
```

Comment:
This is a module script, means it has own namespace, so declaring temporary vectors in that namespace will make it easier to reference within the script, and wont need to write `const fromWorldPoint = OrbitCameraInputMouse.fromWorldPoint;` later.

Reply:
Yep, some of these are scoped to a class  ie `OrbitCamera` and `OrbitCameraInputMouse`